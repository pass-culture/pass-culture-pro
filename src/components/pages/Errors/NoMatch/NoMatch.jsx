import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'

import Icon from 'components/layout/Icon'

const NoMatch = ({ redirect }) => (
  <main className="page fullscreen no-match">
    <Icon svg="ico-404" />
    <div className="nm-title">
      {'Oh non !'}
    </div>
    <div className="nm-subtitle">
      {"Cette page n'existe pas."}
    </div>
    <Link
      className="nm-redirection-link"
      to={redirect}
    >
      {"Retour Ã  la page d'accueil"}
    </Link>
  </main>
)

NoMatch.defaultProps = {
  redirect: '/accueil',
}

NoMatch.propTypes = {
  redirect: PropTypes.string,
}

export default NoMatch
