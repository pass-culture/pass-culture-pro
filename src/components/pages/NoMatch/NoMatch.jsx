import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Redirect } from 'react-router-dom'

class NoMatch extends PureComponent {
  constructor(props) {
    super(props)
    this.timer = null
    this.state = { timing: 5 }
  }

  componentDidMount() {
    const timeout = 1000
    this.timer = setInterval(() => {
      this.setState(({ timing }) => ({ timing: timing - 1 }))
    }, timeout)
  }

  componentWillUnmount() {
    if (this.timer) clearInterval(this.timer)
    this.timer = null
  }

  renderRedirect = () => <Redirect to="/" />
  renderRedirecting = () => <span>{'Redirecting...'}</span>

  renderTimer = timing => <span>{`You will be redirect in ${timing} seconds`}</span>

  render() {
    const { timing } = this.state
    const { location } = this.props
    if (timing < 0) return this.renderRedirect()
    return (
      <div id="page-redirect">
        <h3 className="title">{`404 Not found ${location.pathname}`}</h3>
        <p className="content">
          {timing > 0 && this.renderTimer(timing)}
          {timing === 0 && this.renderRedirecting()}
        </p>
      </div>
    )
  }
}

NoMatch.propTypes = {
  location: PropTypes.shape().isRequired,
}

export default NoMatch
