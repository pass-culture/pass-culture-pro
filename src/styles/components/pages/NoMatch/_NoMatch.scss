.no-match {
  background-image: $linear-gradient-to-bottom-right;
  width: 100%;
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: center;
  color: $white;
  font-weight: $weight-medium;

  h1 {
    font-size: rem(40px);
    padding-top: rem(32px);
  }

  .subtitle {
    font-size: rem(24px);
    font-style: italic;
    padding-top: rem(22px);
  }

  .redirection-info{
    align-self: center;
    bottom: rem(46px);
    font-size: rem(16px);
    padding-top: rem(16px);
    position: absolute;
  }
}
