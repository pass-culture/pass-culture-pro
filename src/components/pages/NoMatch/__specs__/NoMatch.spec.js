import React from 'react'
import { shallow } from 'enzyme'
import NoMatch from '../NoMatch'

describe('src | components | pages | NoMatch', () => {
  let props

  beforeEach(() => {
    props = {
      location: {
        pathname: '/fake-url'
      }
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<NoMatch {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })
})
