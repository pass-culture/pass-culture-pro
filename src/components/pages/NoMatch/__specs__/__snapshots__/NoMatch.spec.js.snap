// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | NoMatch should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoMatch
    location={
      Object {
        "pathname": "/fake-url",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="title"
        >
          404 Not found /fake-url
        </h3>,
        <p
          className="content"
        >
          <span>
            You will be redirect in 5 seconds
          </span>
        </p>,
      ],
      "id": "page-redirect",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404 Not found /fake-url",
          "className": "title",
        },
        "ref": null,
        "rendered": "404 Not found /fake-url",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              You will be redirect in 5 seconds
            </span>,
            false,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You will be redirect in 5 seconds",
            },
            "ref": null,
            "rendered": "You will be redirect in 5 seconds",
            "type": "span",
          },
          false,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="title"
          >
            404 Not found /fake-url
          </h3>,
          <p
            className="content"
          >
            <span>
              You will be redirect in 5 seconds
            </span>
          </p>,
        ],
        "id": "page-redirect",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404 Not found /fake-url",
            "className": "title",
          },
          "ref": null,
          "rendered": "404 Not found /fake-url",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                You will be redirect in 5 seconds
              </span>,
              false,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You will be redirect in 5 seconds",
              },
              "ref": null,
              "rendered": "You will be redirect in 5 seconds",
              "type": "span",
            },
            false,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
